R/waapwlspetpeese.R

```diff
@@ -0,0 +1,610 @@
+  effectTest <- createJaspTable(title = gettext("Test of Effect"))
+  effectTest$position <- 1
+  fitTests[["effectTest"]] <- effectTest
+
+  effectTest$addColumnInfo(name = "type",  title = "",            type = "string")
+  effectTest$addColumnInfo(name = "stat",  title = gettext("t"),  type = "number")
+  effectTest$addColumnInfo(name = "df",    title = gettext("df"), type = "integer")
+  effectTest$addColumnInfo(name = "pVal",  title = gettext("p"),  type = "pvalue")
+
+  if (!is.null(models)) {
+
+    # WLS
+    if (type == "waapWls") {
+      effectTest$addRows(list(
+        type = "WLS",
+        stat = models[["wls"]]["(Intercept)", "t value"],
+        df   = attr(models[["wls"]], "df"),
+        pVal = models[["wls"]]["(Intercept)", "Pr(>|t|)"]
+      ))
+
+      # WAAP
+      .wwppAddWaapFootnote(effectTest, models[["waap"]])
+      if (length(models[["waap"]]) == 0)
+        effectTest$addRows(list(type = "WAAP"))
+      else
+        effectTest$addRows(list(
+          type = "WAAP",
+          stat = models[["waap"]]["(Intercept)", "t value"],
+          df   = attr(models[["waap"]], "df"),
+          pVal = models[["waap"]]["(Intercept)", "Pr(>|t|)"]
+        ))
+    } else if (type == "petPeese") {
+      # PET
+      effectTest$addRows(list(
+        type = "PET",
+        stat = models[["pet"]]["(Intercept)", "t value"],
+        df   = attr(models[["pet"]], "df"),
+        pVal = models[["pet"]]["(Intercept)", "Pr(>|t|)"]
+      ))
+    }
+
+  }
+
+
+  ### test of bias
+  if (type == "petPeese") {
+    biasTest <- createJaspTable(title = gettext("Test of Publication Bias"))
+    biasTest$position <- 2
+    fitTests[["biasTest"]] <- biasTest
+
+    biasTest$addColumnInfo(name = "type",  title = "",                type = "string")
+    biasTest$addColumnInfo(name = "stat",  title = gettext("t"),      type = "number")
+    biasTest$addColumnInfo(name = "df",    title = gettext("df"),     type = "integer")
+    biasTest$addColumnInfo(name = "pVal",  title = gettext("p"),      type = "pvalue")
+
+    if (!is.null(models)) {
+
+      biasTest$addRows(list(
+        type = "PET",
+        stat = models[["pet"]]["PET", "t value"],
+        df   = attr(models[["pet"]], "df"),
+        pVal = models[["pet"]]["PET", "Pr(>|t|)"]
+      ))
+
+    }
+  }
+
+  return()
+}
+.wwppMakeEstimatesTables     <- function(jaspResults, dataset, options, type) {
+
+  models   <- jaspResults[["models"]]$object
+
+  ### assuming heterogeneity
+  if (is.null(jaspResults[["estimates"]])) {
+    # create container
+    estimates <- createJaspContainer(title = gettext("Estimates"))
+    estimates$position <- 2
+    estimates$dependOn(c(.wwppDependencies, "estimates"))
+    jaspResults[["estimates"]] <- estimates
+  } else {
+    estimates <- jaspResults[["estimates"]]
+  }
+
+  # mean estimates
+  if (is.null(estimates[["mean"]]) && options[["estimatesMean"]]) {
+    estimatesMean <- createJaspTable(title = gettextf(
+      "Mean Estimates(%s)",
```

Comment:
This could have a space between Estimates and (%s), but maybe this is on purpose.

Reply:
it was not, good catch, thanks!